### DISCUSSION TITLE
v5.2.1

### DISCUSSION THREAD

**Discussion by niden** [phalcon/cphalcon#16299] [General]
<hr /><em>This discussion was created from the release <a href='https://github.com/phalcon/cphalcon/releases/tag/v5.2.1'>v5.2.1</a>.</em>

**Comment by borisdelev** [comment]
I am wonder about Phalcon PHP 8.2 dll for Windows- know there is some issue... but does will be possible in feature or any info?

**Reply by lubber-de** [reply]
> I am wonder about Phalcon PHP 8.2 dll for Windows- know there is some issue... but does will be possible in feature or any info?

See https://github.com/phalcon/cphalcon/issues/16318#issuecomment-1590522445 where hopefully the phalcon developers could get in touch with another developer who is keen to fix php 8.2/8.3 extensions to work on Windows

**Comment by s-ohnishi** [comment]
I installed Phalcon5.2.1 with pecl based on php:8.2-fpm-bullseye with Docker.
The installation itself completed, but a problem occurred.
in .volt `Deprecated: Creation of dynamic property` occurs frequently.
This message pointed out:
`{% if (session.has('auth') and menu['with_auth'] === true) or menu['with_auth'] === false %} `
or
`{% if session.has('auth') %} `

These are converted to:
`<?php if (($this->session->has('auth') && $menu['with_auth'] === true) || $menu['with_auth'] === false) { ?> `
or
`<?php if ($this->session->has('auth')) { ?>`


How can I avoid this?
Is it only possible to wait for the volt compiler to be updated, or to declare properties in the view class or inherit `stdClass`?

**Reply by s-ohnishi** [reply]
This problem originated from [this](https://github.com/phalcon/cphalcon/issues/16263)

**Reply by s-ohnishi** [reply]
I defined `__set()` in the myVoltEngine class that extends `Phalcon\Mvc\View\Engine\Volt` and registered it as an engine for `.volt`.
Then I got another error `Deprecated: Use of “self” in callables is deprecated`.
